src/molecules/Quaternion.h

```diff
@@ -38,7 +38,9 @@ class Quaternion {
 		scale( 1./ s);
 	}
 	void normalize() {
-		scaleinv(sqrt(magnitude2()));
+		if (!isNormalized()) { // GH#267 Avoid ASM `call sqrt` for identity
```

Comment:
I think the benefit of this depends on how often `normalize()` is called and, then, what ratio of quaternions are not normalized. If the majority of the particles needs to be normalized, this check could add more overhead than it saves (due to the calculation effort of `isNormalized()` -> `magnitude2()`).
But I think this is hard to quantify and highly depends on the case (and if spherical or not).

Reply:
`magnitude2` needs to be computed regardless, so storing it to a frame local variable (register) and then running the check would have virtually no additional computational cost, but would duplicate some code here.  
I have not yet looked into how expensive `fabs` is (probably extremely negligible). At any rate, it will be much, much faster than computing the square root of anything other than 1.  
Even though normalization is idempotent, I don't think it makes bad code to check if it is even necessary, but it is indeed the least impactful change here.

Reply:
After trying a few things and looking at the decompiled assembler code, I must conclude that it is probably not worth optimizing this part, since gcc generates a few more instructions than I anticipated including a trip to memory.

Reply:
Ok, I see. So you would recommend to just undo this commit [279c0b3](https://github.com/ls1mardyn/ls1-mardyn/pull/274/commits/279c0b3cf36c739b4ffd072a119e629618e24d75)?